### PR TITLE
Experimental modal component

### PR DESCRIPTION
**Pull request by hachilam92**
Achieved:
1. use "react-modal" package;
2. add empty ''Modal" component in "App" component, the **toggling** and **content loading** of "Modal" is managed the states of "App";
3. the updating methods of the above states are shared within all pages via "ModalContext", so we can change the modal content and trigger it in any place;
4. example page: the "CreateProfile" page can be triggered by the "Make an offer" button in "BrowseTask" page;

Continuing:
1. add more styles on "Modal" ;
2. add embedded buttons on "Modal" (e.g. closing button);
3. more tests;

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -14,22 +15,59 @@ import Landing from './pages/Landing';
...
+    borderRadius: '10px',
+    marginRight: '-50%',
+    transform: 'translate(-50%, -50%)'
+  }
+};
+
+Modal.setAppElement('#root');
 
 const App = () => {
+  const [modalIsOpen, setIsOpen] = useState(false);
```
Comment (KieraDOG) [inline src/App.js:36]: 把这些 state 和 Modal 都放到 Provider 里面去？

**Review by KieraDOG — COMMENTED** [review]
Modal content 这个概念很赞

**Review by ttcode10 — APPROVED** [review]
Approved

**Review by JavaScriptN0ob — APPROVED** [review]

**Comment by hachilam92** [comment]
正在根据龙哥的建议进行修改，完成后会发起新的PR

**Pull request closed without merging**